.github/workflows/integration-tests.yml

```diff
@@ -16,18 +16,24 @@ jobs:
     
     # Run if:
     # 1. Manual trigger or push to main, OR
-    # 2. PR from trusted author (COLLABORATOR), OR
-    # 3. PR from untrusted author with "safe to test" label
+    # 2. PR with "safe-to-test" label added
     if: |
       github.event_name != 'pull_request_target' ||
-      github.event.pull_request.author_association == 'COLLABORATOR' ||
       contains(github.event.pull_request.labels.*.name, 'safe-to-test')
     
     permissions:
       id-token: write
       contents: read
+      pull-requests: write
     
     steps:
+    - name: Remove safe-to-test label after use
```

Comment:
One issue with doing this is that this will prevent re-driving integration workflows that fail due to causes outside of the code. Say there's an AWS outage while a workflow gets run, the label will get removed, then it will have to get re-added so that a new PR workflow can get triggered on the same commit, when in fact that relabeling should not be necessary from a safety standpoint, the commit has already been approved for integ tests by a person.

One way to get around this is to remove the safe-to-test label on the `synchronize` event like [here](https://github.com/aws/secrets-store-csi-driver-provider-aws/pull/539/files#diff-5a954d9963668387bf1e8bf9b2263a657b7acae10dd6b72a14d2b5ca2bd073e2)